File: app/src/main/java/mx/yellowme/app/simple/UmbrellaPresenter.java
package mx.yellowme.app.simple;

import mx.yellowme.waterproof.mvp.simple.SimplePresenter;
import mx.yellowme.waterproof.mvp.simple.SimpleView;

/**
 * Created by migdonio on 8/1/17.
 */

public class UmbrellaPresenter
        extends SimplePresenter<Umbrella> {
    public UmbrellaPresenter(SimpleView mView) {
        super(mView);
    }

    @Override
    public void loadElement(boolean forceUpdate) {
        // load by Database or Rest API
        Umbrella umbrella = createDummyUmbrella();
        mView.showElement(umbrella);
    }

    private Umbrella createDummyUmbrella() {
        Umbrella umbrella = new Umbrella();
        umbrella.setColor("Blue");
        umbrella.setPrice(239.99);

        return umbrella;
    };
}

File: app/src/main/java/mx/yellowme/app/simple/UmbrellaPresenter.kt
package mx.yellowme.app.simple

import mx.yellowme.waterproof.mvp.simple.SimplePresenter
import mx.yellowme.waterproof.mvp.simple.SimpleView

/**
 * Created by migdonio on 8/1/17.
 */

class UmbrellaPresenter(mView: SimpleView<*>) : SimplePresenter<Umbrella>(mView) {

    override fun loadElement(forceUpdate: Boolean) {
        // load by Database or Rest API
        val umbrella = createDummyUmbrella()
        mView!!.showElement(umbrella)
    }

    private fun createDummyUmbrella(): Umbrella {
        val umbrella = Umbrella()
        umbrella.color = "Blue"
        umbrella.price = 239.99

        return umbrella
    }
}
